### PR TITLE
feat(alpinejs): allow customizing the Alpine instance

### PR DESCRIPTION
**Pull request by florian-lefebvre**
## Changes

- Allows extending Alpine using the new `entrypoint` configuration. Most of the logic is taken from the vue integration

## Testing

Tested manually, and I also added e2e tests.

## Docs

https://github.com/withastro/docs/pull/6534

/cc @withastro/maintainers-docs for feedback!

### CODE REVIEW

**Review by matthewp — APPROVED** [review]
lgtm! I like that this follows the similar pattern we have with Vue.

**Comment by florian-lefebvre** [comment]
The only thing I'm not sure about is the option name, but it's really easy to update

File: packages/integrations/alpinejs/src/index.ts [hunk 1]
```diff
@@ -1,16 +1,83 @@
 import type { AstroIntegration } from 'astro';
+import type { Plugin } from 'vite';
+import { resolve } from 'node:path';
 
-export default function createPlugin(): AstroIntegration {
+interface Options {
+	entrypoint?: string;
```
Comment (ematipico) [inline packages/integrations/alpinejs/src/index.ts:6]: Can we document this option?

**Review by ematipico — APPROVED** [review]

File: packages/integrations/alpinejs/src/index.ts (hunk 1 above)
Comment (florian-lefebvre) [reply packages/integrations/alpinejs/src/index.ts:6]: With JSDoc you mean? FYI the vue integration does not document options with JSDoc so I didn't do it but I can!

Comment (ematipico) [reply packages/integrations/alpinejs/src/index.ts:6]: Yeah with JSDoc, so users can see the description of the options (I think)

**Review by MoustaphaDev — APPROVED** [review]
lgtm!

**Review by ematipico — APPROVED** [review]

**Comment by ematipico** [comment]
Ah shoot, I didn't notice that we requested a review from docs 😱 sorry!

**Pull request merged**